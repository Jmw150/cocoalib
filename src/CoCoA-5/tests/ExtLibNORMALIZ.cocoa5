NmzComputation;

use QQ[x,y,z];

sorted(NmzNormalToricRing([x^3, x^2*y, y^3]));
sorted(NmzIntClosureToricRing([x^3, x^2*y, y^3]));
sorted(NmzIntClosureMonIdeal([x^3, x^2*y, y^3, x*y^2]));
sorted(NmzIntClosureMonIdeal([x^3, x^2*y, y^3, x*y^2],z));
sorted(NmzEhrhartRing([x^3, x^2*y, y^3, x*y^2],z));

// testing HilbertSeries and QuasiPolynomial
M := mat(ZZ, [ [0,  2,  1], [0, -2,  3], [2, -2,  3] ]);
InhomIneqInput := record[ inhom_inequalities := M, grading := mat([[1,0]]) ];
InhomIneq := NmzComputation(InhomIneqInput, ["HilbertSeries"]);
HS := InhomIneq.HilbertSeries;
HQ := InhomIneq.HilbertQuasiPolynomial;
HS;
HQ;
EvalQuasiPoly(HQ,0);
EvalQuasiPoly(HQ,151);

// test period > 1
Cinput := record[integral_closure := M, grading := mat([[1,1,1]]) ];
C := NmzComputation(Cinput, ["HilbertSeries"]);
C.HilbertSeries;
CHQ := C.HilbertQuasiPolynomial;
CHQ;
EvalQuasiPoly(CHQ,0);
EvalQuasiPoly(CHQ,151);
