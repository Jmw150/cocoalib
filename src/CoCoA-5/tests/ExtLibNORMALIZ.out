<built-in function>
[y^3,  x*y^2,  x^2*y,  x^3]
[y,  x]
[y^3,  x*y^2,  x^2*y,  x^3]
[y,  x,  y^3*z,  x*y^2*z,  x^2*y*z,  x^3*z]
[y^3*z,  x*y^2*z,  x^2*y*z,  x^3*z]
record[DenFactors := record[RemainingFactor := 1, factors := [-t +1,  t], multiplicities := [1,  1]], num := t +1]
[2]
2
2
record[DenFactors := record[RemainingFactor := 1, factors := [-t +1,  -t^3 +1], multiplicities := [1,  2]], num := t^5 +2*t^4 +5*t^3 +5*t^2 +2*t +1]
[(8/9)*t^2 +2*t +1,  (8/9)*t^2 +(14/9)*t +5/9,  (8/9)*t^2 +(16/9)*t +8/9]
1
20503
